### PR TITLE
Add support for setting / modifying all fields of a form at once

### PR DESCRIPTION
**Pull request by thomashoneyman**
## What does this pull request do?

Solves #31 by adding `SetAll` and `ModifyAll` queries, so a user can provide a new set of input values to overwrite the current ones, or a new set of input functions to modify existing inputs. 

In addition, Formless now exports query / action functions for every query, instead of only the `modify` / `validate` variants, so a user can use `F.setAll_`, `F.reset_`, etc, and adds documentation for each one. This is because a) it's necessary to provide some helpers for creating queries for things like `Modify` which require injecting a variant; and b) there are more helper queries than actual queries in the component, to keep the component simpler. There was also some confusion in Slack recently over why some queries were just data constructors and others were functions.

## Other Notes:

As implemented, I require that you use `wrapInputFields` or `wrapInputFunctions` to wrap your new record of inputs / input functions rather than supply a plain record. This is due to needing to prove that you are providing a `form Record InputField` or `form Record InputFunction` that is identical to the current form in the component.

Example:

```purescript
H.query unit $ F.setValidateAll_ $ F.wrapInputFields { foo: 1, bar: "x", baz: false }
H.query unit $ F.modifyAll_ $ F.wrapInputFunctions { foo: (+) 1, bar: (<>) "x", baz: not }
```

### CODE REVIEW

File: src/Formless/Query.purs [hunk 1]
```diff
@@ -166,16 +170,148 @@ validate
...
+andThen = AndThen
+
+-- | `andThen` as an action, so you don't need to specify a `Unit`
+-- | result. Use to skip a use of `Halogen.action`.
+andThen_
+  :: ∀ pq cq cs form m
+	 . Query pq cq cs form m Unit
+  -> Query pq cq cs form m Unit
+  -> Query pq cq cs form m Unit
+andThen_ a b = AndThen a b unit
```
Comment (davezuch) [inline src/Formless/Query.purs:317]: https://pursuit.purescript.org/packages/purescript-birds/1.0.4/docs/Aviary.Birds#v:robinstar

Comment (davezuch) [reply src/Formless/Query.purs:317]: I'm just playin

Comment (thomashoneyman) [reply src/Formless/Query.purs:317]: Just absolutely ludicrous. The sort of thing I'd expect from @crcornwell. How did you even find that?

**Comment by dariooddenino** [comment]
I'm not sure if I missed something, but I couldn't find `setValidateAll` and `modifyValidateAll` anywhere! :smile:

```F.setAll_ $ F.wrapInputFields { ... } `F.andThen_` F.validateAll_``` is still better than what I was doing before anyways though!

**Comment by thomashoneyman** [comment]
@dariooddenino 🤦‍♂️ I didn't include it as I was working because you could easily do it with the code you wrote above. But then I forgot to include the query helper at the end, which would simply combine the two like that!

```purescript
setValidateAll_ fields = andThen_ (setAll_ fields) validateAll_
```

**Pull request merged**